File: test-files/vulnerable.py
"""
Test file with intentional security vulnerabilities
Used to test SemgrepGuard extension
"""
import os
import sqlite3
import subprocess
import hashlib

# SQL Injection Vulnerability
def get_user(user_id):
    conn = sqlite3.connect('users.db')
    cursor = conn.cursor()
    # VULNERABLE: SQL injection
    query = f"SELECT * FROM users WHERE id = {user_id}"
    cursor.execute(query)
    return cursor.fetchone()

# Command Injection Vulnerability
def run_command(filename):
    # VULNERABLE: Command injection
    os.system(f"cat {filename}")

# Hardcoded Secret
API_KEY = "sk-1234567890abcdef"  # VULNERABLE: Hardcoded secret
PASSWORD = "admin123"  # VULNERABLE: Hardcoded password

# Weak Hashing
def hash_password(password):
    # VULNERABLE: Using MD5 for password hashing
    return hashlib.md5(password.encode()).hexdigest()

# Insecure Deserialization
import pickle
def load_data(data):
    # VULNERABLE: Arbitrary code execution via pickle
    return pickle.loads(data)

# Path Traversal
def read_file(filename):
    # VULNERABLE: Path traversal
    with open(f"/app/data/{filename}", 'r') as f:
        return f.read()

# Subprocess Shell Injection
def process_input(user_input):
    # VULNERABLE: Shell injection
    subprocess.call(f"echo {user_input}", shell=True)

# Debug Mode Left On (example pattern)
DEBUG = True  # Should be False in production

if __name__ == "__main__":
    print("Test file for SemgrepGuard")

File: test-files/vulnerable.js
// Test file with intentional security vulnerabilities
// Used to test SemgrepGuard extension

const express = require('express');
const mysql = require('mysql');
const exec = require('child_process').exec;

const app = express();

// SQL Injection Vulnerability
app.get('/user', (req, res) => {
    const userId = req.query.id;
    // VULNERABLE: SQL injection
    const query = `SELECT * FROM users WHERE id = ${userId}`;
    connection.query(query, (err, results) => {
        res.json(results);
    });
});

// Command Injection
app.get('/ping', (req, res) => {
    const host = req.query.host;
    // VULNERABLE: Command injection
    exec(`ping -c 1 ${host}`, (err, stdout) => {
        res.send(stdout);
    });
});

// XSS Vulnerability
app.get('/greet', (req, res) => {
    const name = req.query.name;
    // VULNERABLE: Reflected XSS
    res.send(`<h1>Hello ${name}</h1>`);
});

// Hardcoded Secret
const JWT_SECRET = "super-secret-key-123";  // VULNERABLE

// Path Traversal
app.get('/file', (req, res) => {
    const filename = req.query.name;
    // VULNERABLE: Path traversal
    res.sendFile(`/app/files/${filename}`);
});

// Insecure Crypto
const crypto = require('crypto');
function encrypt(text) {
    // VULNERABLE: Using weak algorithm
    const cipher = crypto.createCipher('des', 'password');
    return cipher.update(text, 'utf8', 'hex') + cipher.final('hex');
}

app.listen(3000);
